From 156ddafab9969337ec8f080ecd6f637e6305220d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Daniel=20Grayson?= <josedanielgrayson@proton.me>
Date: Thu, 29 May 2025 14:16:24 -0500
Subject: [PATCH] fix: use builtin filereadable function instead of File.exists

---
 lua/neotest-java/init.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lua/neotest-java/init.lua b/lua/neotest-java/init.lua
index 743ce05..4aa3449 100644
--- a/lua/neotest-java/init.lua
+++ b/lua/neotest-java/init.lua
@@ -15,7 +15,7 @@ local junit_version = ch.config().default_version
 local detect_project_type = require("neotest-java.util.detect_project_type")
 
 local check_junit_jar = function(filepath)
-	local exists, _ = File.exists(filepath)
+	local exists = vim.fn.filereadable(filepath) == 1
 	assert(
 		exists,
 		([[
@@ -55,7 +55,7 @@ local NeotestJavaAdapter = {
 		-- build spec
 		return spec_builder.build_spec(args, project_type, ch.get_context().config)
 	end,
-};
+}
 
 -- on init
 (function()
-- 
2.49.0

